ieee-cdf/ieee-cdf-converter/src/main/java/com/powsybl/ieeecdf/converter/IeeeCdfImporter.java

```diff
@@ -0,0 +1,551 @@
+                .setVoltageLevel2(voltageLevel2Id)
+                .setRatedU1(voltageLevel1.getNominalV() * ieeeCdfBranch.getFinalTurnsRatio())
+                .setRatedU2(voltageLevel2.getNominalV())
+                .setR(ieeeCdfBranch.getResistance() * zb)
+                .setX(ieeeCdfBranch.getReactance() * zb)
+                .setG(0)
+                .setB(ieeeCdfBranch.getChargingSusceptance() / zb)
+            .add();
+    }
+
+    private static Terminal getRegulatingTerminal(IeeeCdfBranch ieeeCdfBranch, TwoWindingsTransformer transformer) {
+        Terminal regulatingTerminal = null;
+        if (ieeeCdfBranch.getSide() != null) {
+            switch (ieeeCdfBranch.getSide()) {
+                case CONTROLLED_BUS_IS_ONE_OF_THE_TERMINALS:
+                    int controlBusNum = ieeeCdfBranch.getControlBusNumber();
+                    if (controlBusNum != 0) {
+                        String controlBusId = getBusId(controlBusNum);
+                        if (controlBusId.equals(transformer.getTerminal1().getBusBreakerView().getBus().getId())) {
+                            regulatingTerminal = transformer.getTerminal1();
+                        } else if (controlBusId.equals(transformer.getTerminal2().getBusBreakerView().getBus().getId())) {
+                            regulatingTerminal = transformer.getTerminal2();
+                        } else {
+                            throw new UnsupportedOperationException("Remote control bus not yet supported: " + transformer.getId());
+                        }
+                    }
+                    break;
+                case CONTROLLED_BUS_IS_NEAR_THE_TAP_SIDE:
+                    regulatingTerminal = transformer.getTerminal1();
+                    break;
+                case CONTROLLED_BUS_IS_NEAR_THE_IMPEDANCE_SIDE:
+                    regulatingTerminal = transformer.getTerminal2();
+                    break;
+                default:
+                    throw new IllegalStateException("Unknown branch side: " + ieeeCdfBranch.getSide());
+            }
+        }
+        return regulatingTerminal;
+    }
+
+    private static void createTransformerWithVoltageControl(IeeeCdfBranch ieeeCdfBranch, ContainersMapping containerMapping, PerUnitContext perUnitContext, Network network) {
+        TwoWindingsTransformer transformer = createTransformer(ieeeCdfBranch, containerMapping, perUnitContext, network);
+        boolean regulating = false;
+        double targetV = Double.NaN;
+        Terminal regulatingTerminal = getRegulatingTerminal(ieeeCdfBranch, transformer);
+        if (regulatingTerminal != null) {
+            Bus regulatingBus = regulatingTerminal.getBusView().getBus();
+            if (regulatingBus != null) {
+                regulating = true;
+                targetV = regulatingBus.getV();
+            }
+        }
+        List<Float> rhos = new ArrayList<>();
+        rhos.add(1f); // TODO create full table
```

Comment:
Should we also add a log to display a message to the user that steps are not supported yet?

Reply:
Ok done.